src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -923,27 +923,17 @@ protected override void Generate(DropIndexOperation operation, [CanBeNull] IMode
         protected override void Generate(CreateSequenceOperation operation, IModel model, MigrationCommandListBuilder builder)
         {
             Check.NotNull(operation, nameof(operation));
-            Check.NotNull(builder, nameof(builder));
 
-            if (VersionAtLeast(10, 0))
+            var oldValue = operation.ClrType;
+            if (!VersionAtLeast(10, 0))
             {
                 // "CREATE SEQUENCE name AS type" expression is supported only in PostgreSQL 10 or above.
                 // The base MigrationsSqlGenerator.Generate method generates that expression.
                 // https://github.com/aspnet/EntityFrameworkCore/blob/master/src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs#L533-L535
-                base.Generate(operation, model, builder);
-                return;
+                operation.ClrType = typeof(long);
             }
-
-            // Generate CREATE SEQUENCE without "AS" part.
-            builder
-                .Append("CREATE SEQUENCE ")
-                .Append(_sqlGenerationHelper.DelimitIdentifier(operation.Name, operation.Schema))
-                .Append(" START WITH ")
-                .Append(_typeMappingSource.GetMapping(typeof(long)).GenerateSqlLiteral(operation.StartValue));
-
-            SequenceOptions(operation, model, builder);
-            builder.AppendLine(_sqlGenerationHelper.StatementTerminator);
-            EndStatement(builder);
+            base.Generate(operation, model, builder);
+            operation.ClrType = oldValue;
```

Comment:
This is a bit far-fetched, but I'd rather we didn't set `ClrType` if we don't need to (i.e. when on PG 10.0 or above). In other words, if `VersionAtLeast(10, 0)` just call directly into `base.Generate()` and return. Otherwise looks good, thanks for the correction...